pyatv/airplay/player.py

```diff
@@ -39,42 +42,43 @@ def play_url(self, url, position=0):
 
         resp = None
         try:
-            resp = yield from self.session.post(
+            session = ClientSession(loop=self.loop)
+            resp = yield from session.post(
                 address, headers=headers, data=body, timeout=TIMEOUT)
-            yield from self._wait_for_media_to_end()
+            yield from self._wait_for_media_to_end(session)
         finally:
             if resp is not None:
                 resp.close()
+            # Apple TV 3 Gen have a bug. It stays in playing state forever
+            # after stream finished. We need close session every time when
+            # video stops to fix it.
+            session.close()
```

Comment:
Like this. I use it with HomeAssistant.

```python
import requests
requests.post(
    'http://localhost:8123/api/services/playlist_player/play_media',
    json={
        'entity_id': 'media_player.apple_tv',
        'media_content_type': 'video',
        'media_content_id': 'http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4'
    })
```

I used same logic with my pure python realization:

`requests stream=True` to keep connection open. And `r1.content` at the end to close it.

```
status = 'loading'

r1 = requests.post(
    'http://%s:7000/play' % host,
    data='Content-Location: %s\nStart-Position: 0.0\n\n' % url,
    stream=True
)

while status != 'stopped':
    time.sleep(1)

    r2 = requests.get('http://%s:7000/scrub' % host)

    if status == 'loading':
        if 'position: 0' not in r2.text:
            status = 'playing'
    elif status == 'playing':
        if 'position: 0' in r2.text:
            status = 'stopped'

r1.content
```

But then I switched to HomeAssistant with your library.